### ISSUE TITLE
failed test on `arki_report`

### ISSUE THREAD

**Issue by brancomat** [ARPA-SIMC/arkimet#102]
building arkimet on a f24 (`ventiquattro`) with dballe 7.24:

```
make  check-local
make[4]: Entering directory '/homelocal/makerpm/rpmbuild/BUILD/arkimet-1.4-1/arki'
for test in tests/arki-test; do \
	../arki/runtest $test ; \
done
Moving to test directory /tmp/tmp.QPav6cYyOA
Running /homelocal/makerpm/rpmbuild/BUILD/arkimet-1.4-1/arki/tests/arki-test 
arki_tests: ..................................................
arki_core_time: .......
```
(...)
```
arki_processor: ....
arki_runtime: ...
arki_dataset_http: ..................................................
arki_report: ..................................................

arki_dataset_index_contents.concurrent: St12system_error: reading 1 byte from child process: No such file or directory
  dataset/index/contents-test.cc:285:actual(readHang.waitUntilHung()) == 'H'

arki_dataset_checker_bufr_simple_sqlite.check_issue51: 1 mismatches in maintenance results:
  segment output not matched: rescanned on testds:20/2005.bufr: should be rescanned

  dataset/tests.cc:971:reporter.check(expected)
  dataset-checker-test.cc:134:actual(checker.get()).check_clean(false, false)

2/4595 tests failed
Failed with result 1
Makefile:4175: recipe for target 'check-local' failed
make[4]: *** [check-local] Error 1
make[4]: Leaving directory '/homelocal/makerpm/rpmbuild/BUILD/arkimet-1.4-1/arki'
Makefile:3985: recipe for target 'check-am' failed
make[3]: *** [check-am] Error 2
make[3]: Leaving directory '/homelocal/makerpm/rpmbuild/BUILD/arkimet-1.4-1/arki'
Makefile:3837: recipe for target 'check-recursive' failed
make[2]: *** [check-recursive] Error 1
make[2]: Leaving directory '/homelocal/makerpm/rpmbuild/BUILD/arkimet-1.4-1/arki'
Makefile:3988: recipe for target 'check' failed
make[1]: *** [check] Error 2
make[1]: Leaving directory '/homelocal/makerpm/rpmbuild/BUILD/arkimet-1.4-1/arki'
Makefile:611: recipe for target 'check-recursive' failed
make: *** [check-recursive] Error 1
errore: Stato d'uscita errato da /var/tmp/rpm-tmp.W4HYXU (%build)
```

**Comment by spanezz** [comment]
Some time the test suite passes, and some time it fails in a different place.

I ran the test suite 100 times on `ventiquattro` collecting results, and failing tests are all about maintenance of `simple` datasets.

I suspect this is related to file modification times not be up to date in the file system by the time the test starts after setting up the test suite. I am trying adding `fsync` calls on directory descriptors and rerunning the tests multiple times to see if I can alter the distribution of results.

**Comment by spanezz** [comment]
I am getting somewhere: the problem is in the test suite, some of the functions to perform changes in test datasets, so that the maintenance fill find and fix problems, have the side effect of changing file modification time, which is a problem when they run after the second has ticked since the dataset was created.
I have fixed some of them and I am chasing the rest. The good news is that this is not a bug in the dataset maintenance code

**Comment by spanezz** [comment]
Fixed in d2bafd89e983d13b96052d015e27dd58f689495d

**Comment by brancomat** [comment]
Thank you, I just packaged arkimet v1.4-2
